### PR TITLE
feat(cdk: subject): add subject

### LINKED ISSUE #154: [cdk: subject] add cdk subject
# API

- [x] I have searched the [issues](https://github.com/IduxFE/components/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?

自己实现一个类rxjs Subject的事件发布订阅管理

### What does the proposed API look like?

```typescript
declare class Subject<T> {
  public dispatch: (value?: T) => void 
  
  public subscribe: ((value: T) => void) => void
}
```

### PR DESCRIPTION
**Pull request by LaamGinghong**
fix #154

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/IduxFE/components/blob/main/docs/contributing.zh.md#commit)
- [x] Tests for the changes have been added/updated or not needed
- [x] Docs and demo have been added/updated or not needed

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Component style update
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

### CODE REVIEW

File: packages/cdk/subject/src/subject.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export class Subject<T = void> {
+  private observers = new Map<symbol, Observer<T>>()
+
+  dispatch(value: T): void {
+    this.observers.forEach(observer => {
+      observer(value)
+    })
+  }
+
+  subscribe(observer: Observer<T>): Subscription {
```
Comment (danranVm) [inline packages/cdk/subject/src/subject.ts:12]: 建议暴露给订阅者的 Subscription  只能取消自己的订阅。
如果是 Subject 本身，则支持 取消全部订阅

File: packages/cdk/subject/docs/index.zh.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+
+
+## 何时使用
+
+## API
+
+### ix-subject
+
+#### Props
```
Comment (coolyuantao) [inline packages/cdk/subject/docs/index.zh.md:17]: 哪来的`props`，哈哈

File: packages/cdk/subject/demo/basic.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  en: Basic usage
+---
+
+## zh
+
+## en
+
+## demo
+
+```html
```
Comment (coolyuantao) [inline packages/cdk/subject/demo/basic.md:14]: demo改下吧

**Review by coolyuantao — CHANGES_REQUESTED** [review]

**Review by danranVm — APPROVED** [review]
LGTM.

**Pull request merged**